### PR TITLE
Fix #69: Build with GHC 9.2

### LINKED ISSUE #69: Compile failure with GHC 9.2
Compilation fails with GHC 9.2 (see below).
There is a fix by @RyanGIScott on head.hackage: https://gitlab.haskell.org/ghc/head.hackage/-/blob/master/patches/hashtables-1.2.4.1.patch

```
$ cabal install -w ghc-9.2.0 hashtables

Build profile: -w ghc-9.2.0.20210806 -O1
In order, the following will be built (use -v for more details):
 - hashtables-1.2.4.1 (lib) (requires build)
...
[ 3 of 12] Compiling Data.HashTable.Internal.IntArray ( src/Data/HashTable/Internal/IntArray.hs, dist/build/Data/HashTable/Internal/IntArray.o, dist/build/Data/HashTable/Internal/IntArray.dyn_o )

src/Data/HashTable/Internal/IntArray.hs:55:18: error:
    • Couldn't match type ‘Word16#’ with ‘Word#’
      Expected: Word# -> Elem
        Actual: Word16# -> Word16
    • In the expression: W16#
      In an equation for ‘primWordToElem’: primWordToElem = W16#
   |
55 | primWordToElem = W16#
   |                  ^^^^

src/Data/HashTable/Internal/IntArray.hs:66:34: error:
    • Couldn't match expected type ‘Word#’ with actual type ‘Word16#’
    • In the first argument of ‘word2Int#’, namely ‘w#’
      In the expression: word2Int# w#
      In an equation for ‘elemToInt#’:
          elemToInt# (W16# w#) = word2Int# w#
   |
66 | elemToInt# (W16# w#) = word2Int# w#
   |                                  ^^
Error: cabal: Failed to build hashtables-1.2.4.1.
```

Blocking:
- Agda https://github.com/agda/agda/issues/5442

### PR DESCRIPTION
**Pull request by andreasabel**
Fix #69: Build with GHC 9.2

Also:
- Add missing modules to testsuite entry in cabal file
- Silence import warnings
- Haskell-CI on Github actions for 7.8 - 9.2
  Successful run at https://github.com/andreasabel/hashtables/actions/runs/1405477638.
  Will run here once you merge into `master`.
- Updated `.travis.yml`
- Bump patch version

### CODE REVIEW

**Comment by patrickt** [comment]
Appreciate the patch—this is blocking `hedgehog` from upgrading.

File: src/Data/HashTable/Internal/CheapPseudoRandomBitStream.hs [hunk 1]
```diff
@@ -15,10 +15,9 @@ import           Data.STRef
 import           Data.Vector.Unboxed           (Vector)
 import qualified Data.Vector.Unboxed           as V
 
-#if __GLASGOW_HASKELL__ >= 808
 import           Data.Word                     (Word32, Word64)
-#else
-import           Data.Word                     (Word, Word32, Word64)
+#if !MIN_VERSION_base(4,8,0)
```
Comment (erikd) [inline src/Data/HashTable/Internal/CheapPseudoRandomBitStream.hs:19]: :+1:

**Review by erikd — APPROVED** [review]

**Comment by erikd** [comment]
Version 1.2.4.2 uploaded to Hackage.

**Comment by andreasabel** [comment]
Maybe push tag `v1.2.4.2` (atm missing from https://github.com/gregorycollins/hashtables/tags).

**Comment by erikd** [comment]
Tag pushed. Will update the CHANGELOG as well, but the official release is on Hackage and cannot be changed.

**Pull request merged**